sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientTelemetryConfig.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.core.http.ProxyOptions;
+import com.azure.cosmos.ClientTelemetryConnectionConfig;
+
+import java.time.Duration;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public class ClientTelemetryConfig {
+    private final boolean clientTelemetryEnabled;
+    private final Duration httpNetworkRequestTimeout;
+    private final int maxConnectionPoolSize;
+    private final Duration idleHttpConnectionTimeout;
+    private final ProxyOptions proxy;
+
+    public ClientTelemetryConfig(boolean clientTelemetryEnabled, ClientTelemetryConnectionConfig connectionConfig) {
+        checkNotNull(connectionConfig, "Argument 'connectionConfig' should not be null");
```

Comment:
Even if it is null, we can switch back to defaultConfig API?

Reply:
whenever we reach here, the connectionConfig will not be null, the fallback to default will happen earlier. so want to keep single point decision path.